### PR TITLE
Move predication from Boogie to GPUVerify

### LINKED ISSUE #53: Move predication from Boogie project into GPUVerify
Predication has lived in the Boogie project in case it would be useful for projects other than GPUVerify. However, this has not proved to be the case, and as part of a clean-up of Boogie it makes sense to remove predication from the Boogie project and make it part of GPUVerify.

### PR DESCRIPTION
**Pull request by afd**
Fixes #53.

### CODE REVIEW

File: GPUVerifyVCGen/SmartBlockPredicator.cs [hunk 1]
```diff
@@ -0,0 +1,818 @@
+namespace GPUVerify
```
Comment (jeroenk) [inline GPUVerifyVCGen/SmartBlockPredicator.cs:1]: Does this file need the usual banner we have on all other files?

File: GPUVerifyVCGen/UniformityAnalyser.cs [hunk 2]
```diff
@@ -0,0 +1,579 @@
+﻿namespace GPUVerify
```
Comment (jeroenk) [inline GPUVerifyVCGen/UniformityAnalyser.cs:1]: Does this file need the usual banner we have on all other files?

**Review by jeroenk — COMMENTED** [review]
Just two small comments, as this is mostly a 1-to-1 copy of code that has existed for a while.

File: GPUVerifyVCGen/SmartBlockPredicator.cs (hunk 1 above)
Comment (afd) [reply GPUVerifyVCGen/SmartBlockPredicator.cs:10]: Done.

File: GPUVerifyVCGen/UniformityAnalyser.cs (hunk 2 above)
Comment (afd) [reply GPUVerifyVCGen/UniformityAnalyser.cs]: Done.

**Pull request merged**